lib/src/ga_client.dart

```diff
@@ -0,0 +1,53 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+import 'constants.dart';
+
+class GAClient {
+  final String measurementId;
+  final String apiSecret;
+  final String postUrl;
+  final http.Client _client;
+
+  GAClient({
+    required this.measurementId,
+    required this.apiSecret,
+  })  : postUrl =
+            '$kAnalyticsUrl?measurement_id=$measurementId&api_secret=$apiSecret',
+        _client = http.Client();
```

Comment:
should we allow tests to provide their own (fake) client?

Reply:
This could be useful, but I'm not sure how helpful it will be because the GA servers always send 2xx status codes, even if the data is malformed. Additionally, the server doesn't send back any payload. Open to discuss this further

Reply:
Ahh, I didn't realize they always return 2xx, lol. I think there's still value in providing a fake client as it means in tests you can verify the request looks correct.